lib/arborist/reify.js

```diff
@@ -446,7 +448,20 @@ module.exports = cls => class Reifier extends cls {
     process.emit('time', timer)
     this.addTracker('reify', node.name, node.location)
 
+    const { npmVersion, nodeVersion } = this.options
     const p = Promise.resolve()
+      .then(() => {
```

Comment:
this function might be a bit cleaner if you use `async [_reifyNode] (node) {`, and then use `await` instead of all the nested `.then`s?

Reply:
it might, but to be consistent with the rest of this code base i left it as-is. we should refactor this repo to be more await-y at some point